libraries/Update/src/HttpsOTAUpdate.cpp

```diff
@@ -0,0 +1,80 @@
+/* Cloud task
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+#include <stdio.h>
+#include <string.h>
+
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "esp32-hal-log.h"
+#include "esp_http_client.h"
+#include "esp_https_ota.h"
+
+#include"HttpsOTAUpdate.h"
+#include"Esp.h"
+#define stackSize 9216
+ 
+static TaskHandle_t ota_task_handle = NULL;
+static const char * server_cert;
+static const char * ota_url;
+static bool skip_cert_common_name_check;
+static void (*http_event_cb)(HttpEvent_t*);
+static int status = -1;
+
+esp_err_t http_event_handler(esp_http_client_event_t *event)
+{
+    http_event_cb(event); 
+    return ESP_OK;
+}
+
+void aws_iot_task(void *param)
+{
+    esp_http_client_config_t config;
+    config.url = ota_url;
+    config.cert_pem = server_cert;
+    config.skip_cert_common_name_check = skip_cert_common_name_check;
+    config.event_handler = http_event_handler;
+    esp_err_t ret = esp_https_ota(&config);
+    if(ret == ESP_OK){
+        status = 1;
+    } else {
+        status = 0;
+        if(ota_task_handle != NULL) {
+            vTaskDelete(ota_task_handle);
```

Comment:
You can just pass `NULL` here if you are executing in the context of the same task that needs to be deleted